### PR TITLE
Implement dynamic manifest libp2p server

### PR DESCRIPTION
**Pull request by masih**
Implement a libp2p server that serves dynamically changing manifests by periodically checking a given manifest path for change. Once change is detected it then uses the manifest sender to publish the new manifest version on pubsub.

Integrate the manifest server into `f3` CLI via command `f3 manifest serve`.

### CODE REVIEW

File: cmd/f3/manifest.go [hunk 1]
```diff
@@ -47,6 +55,155 @@ var manifestGenCmd = cli.Command{
...
+
+		checkTicker := time.NewTicker(c.Duration("checkInterval"))
+		for c.Context.Err() == nil {
+			select {
+			case <-c.Context.Done():
+				return c.Context.Err()
+			case <-checkTicker.C:
+				if nextManifest, nextManifestVersion, err := loadManifestAndVersion(); err != nil {
+					_, _ = fmt.Fprintf(c.App.ErrWriter, "Failed reload manifest: %v\n", err)
+				} else if manifestVersion != nextManifestVersion {
```
Comment (Kubuxu) [inline cmd/f3/manifest.go:196]: Do we want to force manifest updates to be strictly increasing by 1?

**Review by Kubuxu — APPROVED** [review]
SGTM

File: cmd/f3/manifest.go (hunk 1 above)
Comment (masih) [reply cmd/f3/manifest.go:196]: discussed in persion: varsion is hex encoded hash of content

**Review by adlrocha — APPROVED** [review]
LGTM

**Pull request merged**